File: examples/src/main/scala/examples/nodejs/StringDecoderExample.scala
package examples.nodejs

import com.github.ldaniels528.meansjs.nodejs.buffer.Buffer
import com.github.ldaniels528.meansjs.nodejs.{Require, StringDecoder}
import org.scalajs.dom.console

import scala.scalajs.js

/**
  * String Decoder Example
  * @author lawrence.daniels@gmail.com
  */
class StringDecoderExample(require: Require) {

  val StringDecoder = require[StringDecoder]("string_decoder").StringDecoder
  val decoder = StringDecoder("utf8")

  val cent = Buffer(js.Array(0xC2, 0xA2))
  console.log(decoder.write(cent))

  val euro = Buffer(js.Array(0xE2, 0x82, 0xAC))
  console.log(decoder.write(euro))

}

File: examples/src/main/javascript/nodejs/io/StringDecoderExample.js
/**
 * String Decoder Example
 * @author lawrence.daniels@gmail.com
 */
(function () {

    const StringDecoder = require('string_decoder').StringDecoder;
    const decoder = new StringDecoder('utf8');

    const cent = new Buffer([0xC2, 0xA2]);
    console.log(decoder.write(cent));

    const euro = new Buffer([0xE2, 0x82, 0xAC]);
    console.log(decoder.write(euro));

})();
